Source/Core/VideoCommon/OpcodeDecoding.h

```diff
@@ -110,6 +110,11 @@ class Callback
 
   // Get the current CP state.  Needed for vertex decoding; will also be mutated for CP commands.
   virtual CPState& GetCPState() = 0;
+
+  virtual u32 GetVertexSize(u8 vat)
+  {
+    return VertexLoaderBase::GetVertexSize(GetCPState().vtx_desc, GetCPState().vtx_attr[vat]);
+  }
 #endif
```

Comment:
This doesn't compile when `OPCODE_CALLBACK_USE_INHERITANCE` is not in use, as then `GetVertexSize()` isn't defined. I'm not sure exactly what to do about that though. The easiest fix would be to make `GetVertexSize` pure and just put this in the implementations (in FifoPlayer.cpp, FifoRecorder.cpp, and 2 in FIFOAnalyzer.cpp).

Reply:
Yeah, I did that.